File: 1 C/mario.c
#include <cs50.h>
#include <stdio.h>

void buildPyramid(int bricks);
void buildLine(int brickCount, int total);
void printCharacter(int num, char character);

int main(void)
{
    int bricks = 0;
    while (bricks < 1 || bricks > 8) 
    {
        bricks = get_int("Height: ");
    }

    buildPyramid(bricks);
}

void buildPyramid(int bricks) 
{
    for (int i = 1; i <= bricks; i++)
    {
        buildLine(i, bricks);
    }
}

void buildLine(int brickCount, int total)
{
    if (brickCount > 0)
    {
        int midBuffer = 2;
        int buffer = total - brickCount;
        
        printCharacter(buffer, ' ');
        printCharacter(brickCount, '#');
        printCharacter(midBuffer, ' ');
        printCharacter(brickCount, '#');
        printf("\n");
    }
}

void printCharacter(int num, char character)
{
    for (int i = 0; i < num; i++)
    {
        printf("%c", character);
    }
}

File: 6 Python/mario.py
from cs50 import get_int


def main():
    bricks = 0
    while bricks < 1 or bricks > 8:
        bricks = get_int("Height: ")

    buildPyramid(bricks)


def buildPyramid(bricks):
    for i in range(bricks + 1):
        buildLine(i, bricks)


def buildLine(brickCount, total):
    if brickCount > 0:
        midBuffer = 2
        spaceBuffer = total - brickCount

        printCharacter(spaceBuffer, ' ')
        printCharacter(brickCount, '#')
        printCharacter(midBuffer, ' ')
        printCharacter(brickCount, '#', True)


def printCharacter(num, character, lineEnd=False):
    print(character * num, end="")
    if lineEnd:
        print()


main()
